app/views/Moderation/Reports/index.tsx

```diff
@@ -0,0 +1,223 @@
+            ?.moderatorQuery?.reports?.numberOfSchoolsVerified ?? undefined,
+    };
+
+    const usersPerDistrict = ReportsResponse
+        ?.moderatorQuery?.reports?.usersPerDistrict ?? undefined;
+    const booksOrderedAndIncentivesPerDistrict = ReportsResponse
+        ?.moderatorQuery?.reports?.booksOrderedAndIncentivesPerDistrict ?? undefined;
+    const deliveriesPerDistrict = ReportsResponse
+        ?.moderatorQuery?.reports?.deliveriesPerDistrict ?? undefined;
+
+    const booksPerCategory = ReportsResponse
+        ?.moderatorQuery?.reports?.booksPerCategory ?? undefined;
+    const booksPerGrade = ReportsResponse
+        ?.moderatorQuery?.reports?.booksPerGrade ?? undefined;
+    const booksPerLanguage = ReportsResponse
+        ?.moderatorQuery?.reports?.booksPerLanguage ?? undefined;
+    const booksPerPublisher = ReportsResponse
+        ?.moderatorQuery?.reports?.booksPerPublisher ?? undefined;
+
+    const booksAndCostPerSchool = ReportsResponse
+        ?.moderatorQuery?.reports?.booksAndCostPerSchool ?? undefined;
+
+    const booksPerPublisherPerCategory = ReportsResponse
+        ?.moderatorQuery?.reports?.booksPerPublisherPerCategory ?? undefined;
+
+    const paymentPerOrderWindow = ReportsResponse
+        ?.moderatorQuery?.reports?.paymentPerOrderWindow ?? undefined;
+    const bookGradesPerOrderWindow = ReportsResponse
+        ?.moderatorQuery?.reports?.bookGradesPerOrderWindow ?? undefined;
+
+    const window = ReportsResponse?.orderWindows?.results ?? undefined;
+
+    const publisherColor = booksPerPublisher?.map((publisher) => {
+        const colors = COLORS[Number(publisher?.publisherId) % COLORS.length];
+        return {
+            publisher: publisher?.publisherName,
+            fill: colors,
+        };
+    });
+
+    const orderWindows = window?.map((item) => {
+        const colors = COLORS[Number(item.id) % COLORS.length];
+        return {
+            window: item.title,
+            fill: colors,
+        };
+    });
+
+    return (
+        <>
+            <Container
+                className={styles.reports}
+                heading="Reports"
+                headingSize="small"
+            >
+                <>
+                    <Overall
+                        overallData={overallData}
+                    />
+                    <District
+                        usersPerDistrict={usersPerDistrict}
+                        booksOrderedAndIncentivesPerDistrict={booksOrderedAndIncentivesPerDistrict}
+                        deliveriesPerDistrict={deliveriesPerDistrict}
+                    />
+                    <Window
+                        paymentPerOrderWindow={paymentPerOrderWindow}
+                        bookGradesPerOrderWindow={bookGradesPerOrderWindow}
+                        orderWindows={orderWindows}
+                        booksAndCostPerSchool={booksAndCostPerSchool}
+                    />
+                    {booksPerPublisher && (
```

Comment:
any reason we are making `booksPerPublisher` required here?

Reply:
There was no changes